src/test/repository/SavedPromotion.test.ts

```diff
@@ -0,0 +1,125 @@
+import { getCustomRepository } from 'typeorm';
+import connection from './BaseRepositoryTest';
+import { SavedPromotionRepository } from '../../main/repository/SavedPromotionRepository';
+import { UserRepository } from '../../main/repository/UserRepository';
+import { PromotionRepository } from '../../main/repository/PromotionRepository';
+import { User } from '../../main/entity/User';
+import { Promotion } from '../../main/entity/Promotion';
+import { UserFactory } from '../factory/UserFactory';
+import { PromotionFactory } from '../factory/PromotionFactory';
+import { Discount } from '../../main/entity/Discount';
+import { DiscountFactory } from '../factory/DiscountFactory';
+import { Schedule } from '../../main/entity/Schedule';
+import { ScheduleFactory } from '../factory/ScheduleFactory';
+
+describe('Unit tests for SavedPromotionRepository', function () {
+  let savedPromotionRepository: SavedPromotionRepository;
+  let userRepository: UserRepository;
+  let promotionRepository: PromotionRepository;
+  beforeAll(async () => {
+    await connection.create();
+  });
+
+  afterAll(async () => {
+    await connection.close();
+  });
+
+  beforeEach(async () => {
+    await connection.clear();
+    savedPromotionRepository = getCustomRepository(SavedPromotionRepository);
+    userRepository = getCustomRepository(UserRepository);
+    promotionRepository = getCustomRepository(PromotionRepository);
+  });
+
+  test('Should be able to store a savedpromotion and successfully retrieve the saved promotion', async () => {
+    // make a promotion
+    const user: User = new UserFactory().generate();
+    const discount: Discount = new DiscountFactory().generate();
+    const schedule: Schedule = new ScheduleFactory().generate();
+    const promotion: Promotion = new PromotionFactory().generate(
+      user,
+      discount,
+      [schedule]
+    );
+    await userRepository.save(user);
+    await promotionRepository.save(promotion);
+
+    // save the promotion
+    const saved_user: User = new UserFactory().generate();
```

Comment:
Very minor, but for code style I think you can change this variable to `savedUser`